easy-random-core/src/main/java/org/jeasy/random/ObjenesisObjectFactory.java

```diff
@@ -47,7 +46,7 @@
     @Override
     public <T> T createInstance(Class<T> type, RandomizerContext context) {
         if (context.getParameters().isScanClasspathForConcreteTypes() && isAbstract(type)) {
-            Class<?> randomConcreteSubType = randomElementOf(getPublicConcreteSubTypesOf((type)));
+            Class<?> randomConcreteSubType = context.randomElementOf(getPublicConcreteSubTypesOf((type)));
```

Comment:
We can use a `Random` instance here and initialize it  (only once) with the seed from the context:

```java
    private Random random;

    @Override
    public <T> T createInstance(Class<T> type, RandomizerContext context) {
        if (random == null) {
            random = new Random(context.getParameters().getSeed());
        }
        if (context.getParameters().isScanClasspathForConcreteTypes() && isAbstract(type)) {
            List<Class<?>> publicConcreteSubTypes = getPublicConcreteSubTypesOf(type);
            if (publicConcreteSubTypes.size() == 0) {
                throw new InstantiationError("Unable to find a matching concrete subtype of type: " + type + " in the classpath");
            } else {
                Class<?> randomConcreteSubType = publicConcreteSubTypes.get(random.nextInt(publicConcreteSubTypes.size()));
                return (T) createNewInstance(randomConcreteSubType);
            }
        } else {
            try {
                return createNewInstance(type);
            } catch (Error e) {
                throw new ObjectCreationException("Unable to create an instance of type: " + type, e);
            }
        }
    }
```

Reply:
done